src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4135,6 +4160,26 @@ WorkerLimitCount(MultiExtendedOp *originalOpNode)
 								workerLimitCountInt64)));
 	}
 
+	/* if offset is present, add offset value to limit on workers */
+	if (originalOpNode->limitOffset != NULL)
+	{
+		Const *workerOffsetConst = (Const *) originalOpNode->limitOffset;
+		Const *workerLimitConst = NULL;
+		int64 workerOffsetValueInt64 = DatumGetInt64(workerOffsetConst->constvalue);
+		int64 workerNewLimitValueInt64 = -1;
+
+		/* if a limit is not present, create a new node for limit */
+		if (workerLimitCount == NULL)
+		{
+			workerLimitCount = (Node *) MakeIntegerConstInt64(0);
+		}
+
+		workerLimitConst = (Const *) workerLimitCount;
+		workerNewLimitValueInt64 = DatumGetInt64(workerLimitConst->constvalue);
+		workerNewLimitValueInt64 += workerOffsetValueInt64;
+		workerLimitConst->constvalue = Int64GetDatum(workerNewLimitValueInt64);
+	}
+
```

Comment:
You are correct. I have also changed the code above this part for further simplification.